File: python/resistor-color-duo/resistor_color_duo.py
all_colors = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]

def value(colors):

    if len(colors) < 2:
        raise ValueError("Must supply two values.")

    total = 0

    for i in range(2):

        total *= 10

        if colors[i] in all_colors:
            total += all_colors.index(colors[i])
        else:
            raise ValueError("{} is not a valid color".format(colors[i]))


    return total

File: typescript/resistor-color-duo/resistor-color-duo.ts
export class ResistorColor {

  static readonly AllColors: string[] = [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white',
  ]

  private colors: string[]

  constructor(colors: string[]) {
    if (colors.length < 2) { throw new Error('At least two colors need to be present') }
    this.colors = colors
  }

  value = (): number => {
    return ResistorColor.AllColors.indexOf(this.colors[0]) * 10 + ResistorColor.AllColors.indexOf(this.colors[1])
  }
}
